### PR TITLE
Array shift

### PR DESCRIPTION
**Pull request by egconley**

### CODE REVIEW

**Comment by egconley** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [X] Feature tasks for this challenge are completed
 - [X] Unit tests written and passing
     - [X] “Happy Path” - Expected outcome
     - [X] Expected failure
     - [X] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [X] Picture of whiteboard

File: code401challenges/src/test/java/code401challenges/ArrayShiftTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        int[] expected1 = new int[]{1, 2, 3, 99, 4, 10, 11};
+
+        int[] input2 = new int[]{1, 2, 3, 4, 10, 11, 12};
+        int[] expected2 = new int[]{1, 2, 3, 4, 77, 10, 11, 12};
+
+        int[] result1 = test1.insertShiftArray(input1, 99);
+        Assert.assertArrayEquals(expected1, result1);
+
+        int[] result2 = test1.insertShiftArray(input2, 77);
+        Assert.assertArrayEquals(expected2, result2);
```
Comment (Zaffierce) [inline code401challenges/src/test/java/code401challenges/ArrayShiftTest.java:25]: I like the idea of doing two tests, however, these should be split into their own methods.

public void insertShiftArrayEven() {
//do stuff
}

public void insertShiftArrayOdd() {
//do stuff
}

File: code401challenges/src/main/java/code401challenges/ArrayShift.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            } else if (i == (int) Math.ceil(array.length/2) && array.length %2 == 0) {
+                newArray[i] = value;
+            } else if (i == (int) Math.ceil(array.length/2) && array.length %2 != 0) {
+                newArray[i] = array[i];
+                newArray[i+1] = value;
+            } else if (i > array.length/2 && array.length %2 == 0) {
+                newArray[i] = array[i-1];
+            } else if (i > (array.length/2 + 1) && array.length %2 != 0) {
+                newArray[i] = array[i-1];
+            }
```
Comment (Zaffierce) [inline code401challenges/src/main/java/code401challenges/ArrayShift.java:34]: There's a lot of logic going on here and things are way more complicated than it needs to be.  You can do all of this in 5 lines and by with the use of an additional variable.

```
if (i == midPoint) {
  newArray[i] = value;
} else {
  newArray[i] = array[j];
  j++ }
```
